Add support for new institutions all endpoint

jsatk opened the pull request:
This PR adds support for the new `/institutions/all` and `institutions/all/search` endpoints. 

We were told that as of January 4, 2017 the former `institutions/longtail` and `/institutions/search` endpoints would no longer be supported and we should use the new `/institutions/all` endpoint. We noticed the node module we heavily depend upon does not yet support these. This PR adds support for these endpoints. 

Please let me know if you have any requested changes. I attempted to do this as surgically as possible as to not introduce any inconsistencies. 

This PR addresses https://github.com/plaid/plaid-node/issues/88

jsatk commented:
@michaelckelly Please let me know if this addresses issue https://github.com/plaid/plaid-node/issues/88.

jsatk commented:
It appears this test is flakey and sometimes runs longer than the timeout threshold.  

```
1) Plaid.Client - Auth Plaid.Client.stepAuthUser accepts a send_method in options:
     Error: timeout of 10000ms exceeded. Ensure the done() callback is being called in this test.
```

michaelckelly commented:
@jsatk Thank you for the PR, reviewing now

michaelckelly commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "plaid",
-  "version": "1.1.0",
+  "version": "1.2.0",
```
We have a separate publish process that we use actually -- can you revert this change? I'll publish `1.2.0` once we merge this change.

jsatk replied:
Totally. Will do.

michaelckelly commented on test/authenticated.js:
```diff
@@ -1051,3 +1051,24 @@ describe('Plaid.Client - Longtail Institutions', function() {
+
+  // The /institutions/all endpoint requires additional permissions
+  // to access. Accessing it with the sandbox client ID and secret returns
+  // an error, which we test here.
```
I don't think this comment is accurate given what the code is doing below?

jsatk replied:
This comment is identical to the comment on line 1038–1040. This test is essentially a copy & paste of that test. Was simply attempting to maintain equal test coverage for this new method as was there for the long tail method.

jsatk commented:
@michaelckelly I removed the npm version bump and added a clarifying comment on one of yours. Let me know if there is anything else. Thanks for the review.

jsatk commented:
@michaelckelly Did you have any further requests for this PR? I'd love to see this published soon. If that's not possible I'll publish my fork and use that until this module had support for the new `/institutions/all` endpoint, but I'd prefer to avoid doing that if possible. Let me know if you have any further questions/comments. Thanks!

michaelckelly commented:
Sorry about the delay @jsatk - LGTM!

michaelckelly commented:
@jsatk - published `1.1.1` (just a patch release)
